Formula/rocksdb.rb

```diff
@@ -40,6 +44,9 @@ def install
       -DWITH_ZSTD=ON
     ]
 
+    # ensure generated executables know where to find the rocksdb dylib
+    args << "-DCMAKE_EXE_LINKER_FLAGS=-Wl,-rpath -Wl,#{lib}"
```

Comment:
```suggestion
    args << "-DCMAKE_INSTALL_RPATH=#{lib}"
```
Let's try to be consistent with the way other formulae do this. If that doesn't work, then we can try `CMAKE_INSTALL_NAME_DIR`.

Please squash this change with the first one, and use the preferred commit-message style for homebrew/core, which would be something like: `rocksdb: add head build`

We put the name of the formula first in commit-message headings. Refer to the [documentation](https://docs.brew.sh/Formula-Cookbook#commit) for more details.

Reply:
Hi @carlocab, many thanks for reviewing this PR! 

Apologies about the commit message, I updated it as per your suggestion (using `git commit --amend`). 

Regarding the `cmake` options, I actually tried using either `CMAKE_INSTALL_RPATH` or `CMAKE_INSTALL_NAME_DIR` but unfortunately both do not seem to work. I did not really dig into the RocksDB cmake configuration. Maybe the build step generating the exe utilities do not head those flags (I do not see them when I run `cmake -LAH`)? That is why I resorted to using `DCMAKE_EXE_LINKER_FLAGS`. Do let me know if there is something else I can try instead though!

Reply:
Hm, interesting. Does adding these flags work for you from the stable release, or does it only work on the `HEAD` build?

Reply:
I actually implemented the `HEAD` build because stable does not compile on Apple M1. Only RocksDB master branch has the necessary changes to make it compile on Apple M1 (although I think 6.17.3 that was released 3 days ago addresses this problem).

I did just run `brew install --build-from-source rocksdb` on an Intel Mac and that seems to work fine. So perhaps the problem is caused by brew on the M1 using`/opt/homebrew` instead of `/usr/local`. If I'm not mistaken the RocksDB cmake install script assumes `/usr/local`, so maybe that is why it works there. This is just a guess though as I have not dug into the cmake files to confirm this!

Reply:
Yes, it probably is because of the non-standard prefix. Do you mind testing 6.17.3 if it works for you with these flags, or (preferably) one of `CMAKE_INSTALL_RPATH` or `CMAKE_INSTALL_NAME_DIR`?